pkg/controllers/user/authz/podsecuritypolicy/template.go

```diff
@@ -3,179 +3,131 @@ package podsecuritypolicy
-			_, err := fromTemplateExplicitName(m.policies, m.policyLister, policy.Name, obj)
+			_, err = l.policies.Update(newPolicy)
 			if err != nil {
-				return err
+				return nil, fmt.Errorf("error updating psp: %v", err)
 			}
 		}
 	}
 
-	return nil
+	return obj, nil
 }
-func fromTemplate(policies v1beta12.PodSecurityPolicyInterface, policyLister v1beta12.PodSecurityPolicyLister,
-	key string, originalTemplate *v3.PodSecurityPolicyTemplate) (*v1beta1.PodSecurityPolicy, error) {
-	return fromTemplateExplicitName(policies, policyLister, keyToPolicyName(key), originalTemplate)
-}
-
-func fromTemplateExplicitName(policies v1beta12.PodSecurityPolicyInterface,
-	policyLister v1beta12.PodSecurityPolicyLister, key string,
-	originalTemplate *v3.PodSecurityPolicyTemplate) (*v1beta1.PodSecurityPolicy, error) {
-	template := originalTemplate.DeepCopy()
-
-	objectMeta := v1.ObjectMeta{}
-	objectMeta.Name = key
-	objectMeta.Annotations = make(map[string]string)
-	objectMeta.Annotations[podSecurityTemplateParentAnnotation] = template.Name
-	objectMeta.Annotations[podSecurityVersionAnnotation] = template.ResourceVersion
-
-	psp := &v1beta1.PodSecurityPolicy{
-		TypeMeta: v1.TypeMeta{
-			Kind:       podSecurityPolicy,
-			APIVersion: apiVersion,
-		},
-		ObjectMeta: objectMeta,
-		Spec:       template.Spec,
-	}
-
-	var policy *v1beta1.PodSecurityPolicy
-	var err error
-
-	if !doesPolicyExist(policyLister, key) {
-		policy, err = policies.Create(psp)
-	} else {
-		policy, err = policies.Update(psp)
-	}
 
+func (l *Lifecycle) Remove(obj *v3.PodSecurityPolicyTemplate) (*v3.PodSecurityPolicyTemplate, error) {
+	policies, err := l.policyIndexer.ByIndex(policyByPSPTParentAnnotationIndex, obj.Name)
 	if err != nil {
-		return nil, fmt.Errorf("error creating pod security policy: %v", err)
+		return nil, fmt.Errorf("error getting policies: %v", err)
 	}
 
-	logrus.Debugf("created/updated a pod security policy with name %v", objectMeta.Name)
-
-	return policy, nil
-}
-
-func doesPolicyExist(policyLister v1beta12.PodSecurityPolicyLister, name string) bool {
-	_, err := policyLister.Get("", name)
-
-	return !errors.IsNotFound(err)
-}
-
-func getPodSecurityPolicyTemplateID(projectLister v3.ProjectLister, clusterLister v3.ClusterLister, projectID string,
-	clusterName string) (string, error) {
-	projects, err := projectLister.List("", labels.Everything())
-	if err != nil {
-		return "", fmt.Errorf("error getting projects: %v", err)
-	}
-
-	var project *v3.Project
-
-	for _, candidate := range projects {
-		if candidate.Name == projectID {
-			project = candidate
-			break
+	for _, rawPolicy := range policies {
+		policy := rawPolicy.(*v1beta1.PodSecurityPolicy)
+		err = l.policies.Delete(policy.Name, &v1.DeleteOptions{})
+		if err != nil {
+			return nil, fmt.Errorf("error deleting policy: %v", err)
 		}
 	}
 
-	if project == nil {
-		return "", nil
+	psptpbs, err := l.psptpbIndexer.ByIndex(psptpbByPSPTNameIndex, obj.Name)
```

Comment:
You should not have to delete the PSPTPBs in the mgmt plane here. You did it in your management plane controller here:https://github.com/rancher/rancher/pull/12141/files?diff=split#diff-96038a85f79ba8af9f29050caf441508R46